src/Hint/Match.hs

```diff
@@ -327,13 +329,13 @@ checkSide' x bind = maybe True bool x
       isType "InfixAp" (GHC.LL _ GHC.OpApp{}) = True
       isType "Paren" (GHC.LL _ GHC.HsPar{}) = True
       isType "Tuple" (GHC.LL _ GHC.ExplicitTuple{}) = True
-      isType typ _ = error $ "Hint.Match.checkSide', unknown side condition: '" ++ "is" ++ typ ++ "'"
+      isType _ _ = False
```

Comment:
This seems dodgy

Reply:
Improved this now in two ways : (1) Take into account overloaded literals and (2) the last case now respects the spirit of the original (but likely not the end of the story here - we'll see). Failure count has come down.